src/main/Msgt.razor.cs

```diff
@@ -200,10 +200,47 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
                 // register push notification service worker
                 var objRef = DotNetObjectReference.Create(this);
                 await this.JsRuntime.InvokeVoidAsync("RegisterServiceWorker", objRef);
+
+                //this.instantiatesGoogleDriveImageNeuron = (await neuronQueryService.GetNeurons(
+                //    rq.AvatarUrl,
+                //    new NeuronQuery()
+                //    {
+                //        ExternalReferenceUrl = new string[] { messageTreePluginSettingsService.PosterUrls.InstantiatesGoogleDriveImage }
+                //    }))
+                //    .Items
+                //    .FirstOrDefault();
             }
             await base.OnAfterRenderAsync(firstRender);
         }
 
+        private async void DetectTermianlNeuron()
+        {
+            var neurons = this.Children;
+            if(QueryUrl.TryParse(this.AvatarUrl, out QueryUrl result))
+            {
+                this.instantiatesGoogleDriveImageNeuron = (await this.NeuronQueryService.GetNeurons(
+                result.AvatarUrl,
+                new NeuronQuery()
+                {
+                    ExternalReferenceUrl = new string[] { this.pluginSettingsService.PosterUrls.InstantiatesGoogleDriveImage }
+                }))
+                .Items
+                .FirstOrDefault();
+                foreach (var neuron in neurons)
+                {
+                    var newNeuron = await this.NeuronQueryService.GetNeuronById(
+                        result.AvatarUrl,
+                        this.instantiatesGoogleDriveImageNeuron.Id,
+                        neuron.Neuron.Id,
+                        new NeuronQuery());
+                    if (newNeuron.Count == 1)
```

Comment:
Can we use newNeuron.Single() here instead of ".Count == 1"?

Reply:
this is unable to used